great_expectations/expectations/core/expect_column_proportion_of_non_null_values_to_be_between.py

```diff
@@ -249,24 +261,153 @@ def schema_extra(
-    def _prescriptive_renderer(  # type: ignore[empty-body]  # TODO: remove this with impementation
+    def _prescriptive_renderer(
         cls,
         configuration: Optional[ExpectationConfiguration] = None,
         result: Optional[ExpectationValidationResult] = None,
         runtime_configuration: Optional[dict] = None,
         **kwargs,
-    ) -> list[RenderedStringTemplateContent]: ...
+    ) -> list[RenderedStringTemplateContent]:
+        runtime_configuration = runtime_configuration or {}
+        include_column_name = runtime_configuration.get("include_column_name") is not False
+        styling = runtime_configuration.get("styling")
+        params = (
+            substitute_none_for_missing(
+                configuration.kwargs,
+                [
+                    "column",
+                    "min_value",
+                    "max_value",
+                    "row_condition",
+                    "condition_parser",
+                    "strict_min",
+                    "strict_max",
+                ],
+            )
+            if configuration
+            else {}
+        )
+
+        if params["min_value"] is None and params["max_value"] is None:
+            template_str = "may have any fraction of non-null values."
+        else:
+            at_least_str, at_most_str = handle_strict_min_max(params)
+            if params["min_value"] is None:
+                template_str = f"fraction of non-null values must be {at_most_str} $max_value."
+            elif params["max_value"] is None:
+                template_str = f"fraction of non-null values must be {at_least_str} $min_value."
+            elif params["min_value"] != params["max_value"]:
+                template_str = (
+                    f"fraction of non-null values must be {at_least_str} $min_value "
+                    f"and {at_most_str} $max_value."
+                )
+            else:
+                template_str = "fraction of non-null values must be exactly $min_value."
+
+        if include_column_name:
+            template_str = f"$column {template_str}"
+
+        if params["row_condition"] is not None:
+            (
+                conditional_template_str,
+                conditional_params,
+            ) = parse_row_condition_string_pandas_engine(params["row_condition"])
+            template_str = f"{conditional_template_str}, then {template_str}"
+            params.update(conditional_params)
+
+        return [
+            RenderedStringTemplateContent(
+                content_block_type="string_template",
+                string_template={
+                    "template": template_str,
+                    "params": params,
+                    "styling": styling,
+                },
+            ),
+        ]
+
+    @classmethod
+    @renderer(
+        renderer_type=LegacyDescriptiveRendererType.COLUMN_PROPERTIES_TABLE_DISTINCT_PERCENT_ROW
+    )
+    def _descriptive_column_properties_table_distinct_percent_row_renderer(
```

Comment:
I was unable to find any existing testing patterns for these legacy renderers. I can unit test it, but I'm not sure how that adds much value. I will test this manually with Core Data Docs.